roles/sro_deploy_art_bundle/tasks/main.yml

```diff
@@ -0,0 +1,269 @@
+          -ojsonpath='{.status.conditions[].reason}'
+      register: buildconfig_ready
+      until: "'PodComplete' in buildconfig_ready.stdout"
+      retries: 20
+      delay: 30
+  rescue:
+    - name: "Store logs from build config"
+      shell:
+        oc logs ci-artifacts-builder-1-build
+          -n ci-artifacts
+          > "{{ artifact_extra_logs_dir }}"/failed_builder.log
+
+    - name: "Failing because of previous error"
+      fail: msg="Failing because of build config error"
+
+- name: Build index from latest ART bundle in a pod
+  block:
+    - name: Generate pod resource yaml
+      template:
+        src: "{{ builder_pod_file }}"
+        dest: "{{ artifact_extra_logs_dir }}/builder_pod.yml"
+        mode: 0400
+
+    - name: Create builder pod
+      command: oc apply -f "{{ artifact_extra_logs_dir }}/builder_pod.yml"
+
+    - name: Wait pod finish.
+      command:
+        oc get pod/ci-artifacts-index-builder
+          -n ci-artifacts
+          -ojsonpath='{.status.conditions[].reason}'
+      register: index_builder_ready
+      until: "'PodComplete' in index_builder_ready.stdout"
+      retries: 30
+      delay: 20
+
+  rescue:
+    - name: "Store described index builder"
+      shell:
+        oc describe pod ci-artifacts-index-builder
+          -n ci-artifacts
+          > "{{ artifact_extra_logs_dir }}"/failed_index_builder.desc
+
+    - name: "Store logs from index builder"
+      shell:
+        oc logs ci-artifacts-index-builder
+          -n ci-artifacts
+          > "{{ artifact_extra_logs_dir }}"/failed_index_builder.log
+
+    - name: "Failing because of previous error"
+      fail: msg="Failing because of index builder pod error"
+
+
+- name: Add service account for catalog
+  command: oc apply -f "{{ sro_serviceaccount_file }}"
+
+- name: Add system puller policy
+  command:
+    oc policy add-role-to-user system:image-puller system:serviceaccount:openshift-marketplace:special-resource-operator-catalog
+      -n ci-artifacts
+
+- name: Create SRO catalog from the private index
+  command: oc apply -f "{{ sro_catalogsource_file }}"
+
+- name: Ensure SRO PackageManifest is available
+  block:
+  - name: Wait for the CatalogSource to become available
+    shell:
+      oc get CatalogSource/special-resource-operator-catalog
+        -n openshift-marketplace
+        -oyaml
+        '-ojsonpath={.status.connectionState.lastObservedState}{"\n"}'
+    register: catalog_source_ready
+    until: "'READY' in catalog_source_ready.stdout"
+    retries: 30
+    delay: 30
+
+  - name: Wait for the SRO to be available
+    command:
+      oc get packagemanifests
+        -n openshift-marketplace
+        -l catalog=special-resource-operator-catalog
+    register: sro_package_wait
+    until: sro_package_wait.rc == 0
+    retries: 15
+    delay: 30
+
+  rescue:
+  - name: Mark the failure as flake
+    shell:
+      echo "Failed because of the SRO packagemanifest not available"
+           > "{{ artifact_dir }}/FLAKE"
```

Comment:
I'm not sure this should be considered as a flake, as you're building the catalog yourself?

Reply:
Swapped it with describe+logs.